### PR TITLE
1683-update-heat-agent-queries: Updated heat agent to exclude emissions from p…

### PR DESCRIPTION
**Pull request by nagarajankarthik**
…ower plants in Jurong Island.

### CODE REVIEW

**Comment by nagarajankarthik** [comment]
This is the latest version of the heat emissions agent that is designed to work with industries' data stored in PostgreSQL instead of Blazegraph.

File: Agents/HeatEmissionAgent/HeatEmissionAgent/src/main/java/uk/ac/cam/cares/jps/agent/heat/EndpointConfig.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    private BlazegraphEndpointConfig blazegraphEndpointConfig;
+
+    public EndpointConfig() {
+        ContainerClient containerClient = new ContainerClient();
+        this.blazegraphEndpointConfig = containerClient.readEndpointConfig("blazegraph",
+                BlazegraphEndpointConfig.class);
+        this.postGISEndpointConfig = containerClient.readEndpointConfig("postgis", PostGISEndpointConfig.class);
+    }
+
+    public String getKgurl(String namespace) {
```
Comment (sganguly-cares) [inline Agents/HeatEmissionAgent/HeatEmissionAgent/src/main/java/uk/ac/cam/cares/jps/agent/heat/EndpointConfig.java:18]: Could you standardise the naming of the methods to getKgUrl
https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html

File: Agents/HeatEmissionAgent/HeatEmissionAgent/src/main/java/uk/ac/cam/cares/jps/agent/heat/HeatEmissionAgent.java [hunk 2]
```diff
@@ -1,176 +1,73 @@
...
-			try {
-				fw = new FileWriter(file1);
-				PrintWriter pw = new PrintWriter(fw);
-				pw.println(output);
-				pw.close();
-			} catch (IOException e) {
-				e.printStackTrace();
-			}
+			rdbStoreClient = new RemoteRDBStoreClient(dbUrl, dbUser, dbPassword);
+
```
Comment (sganguly-cares) [inline Agents/HeatEmissionAgent/HeatEmissionAgent/src/main/java/uk/ac/cam/cares/jps/agent/heat/HeatEmissionAgent.java:60]: What about the remaining industries? i.e. pharma, food, semicon and chemicals? do they all come under factory?

File: Agents/HeatEmissionAgent/README.md [hunk 3]
```diff
@@ -1,54 +1,104 @@
...
 
-# Example Heat Emission Data
-Heat Emission Data is returned to the response body in form of a JSON Object which consist of key-value pair. The JSONObject has the 
-key:"result", which contains a JSONArray containing JSONObjects. Each of these individual JSONObjects found within the JSONArray
-provide the heat emission coordinates and the corresponding heat emission value in the unit of MW.
+The heat emission agent calculates heat emissions of data centres and various types of factories. The properties of these industrial facilities that are required to calculate the heat emissions need to be stored in PostgreSQL tables as explained below.
```
Comment (sganguly-cares) [inline Agents/HeatEmissionAgent/README.md:3]: Are the terms 'factories' and 'industrial facilities' interchangeable? According to the Miro board it seems they are different - if they are indeed different then revise the above sentences

File: Agents/HeatEmissionAgent/README.md [hunk 4]
```diff
@@ -1,54 +1,104 @@
...
+#### 1.2 Docker Deployment
+
+- From the same directory location as this README, build the agent's image using the `docker compose build` command. If the container is run as part of a stack, copy the `heatemissionagent.json` file from the `stack-manager-input-config` folder into the `Deploy/stacks/dynamic/stack-manager/inputs/config/services` folder of the stack manager before starting the stack.
+
+
+### 2. Agent route
+
+The agent has a single API route which requires a POST request. It accepts the following input parameters:
+
+- ```dbName```: The name of the PostgreSQL database from which the factories' properties are queried.
```
Comment (sganguly-cares) [inline Agents/HeatEmissionAgent/README.md:55]: 'factories' or 'industrial facilities'?

File: Agents/HeatEmissionAgent/README.md [hunk 5]
```diff
@@ -1,54 +1,104 @@
...
+
+The agent adds a new column called 'heat_emissions' to each of the tables mentioned above containing the heat emissions of each type of industrial facility in units of megawatts.
+
+The following methods are used to estimate the heat emissions of each type of industrial facility:
+
+- Data Centres:
+
+$$ Q = \frac{1.08*P_{max}U_R + 21.53A_F + 9400}{1.0 \times 10^6} $$
+
+The parameters $Q$, $P_{max}$, $U_R$ and $A_F$ denote the heat emissions rate in megawatts, maximum IT capacity in megawatts, utilization rate and gross floor area in square meters respectively.
```
Comment (sganguly-cares) [inline Agents/HeatEmissionAgent/README.md:77]: Minor change - can you write out the parameters as a list rather than as a chunk of text, it is hard to follow

**Review by sganguly-cares — CHANGES_REQUESTED** [review]
Left comments on some of the files.

File: Agents/HeatEmissionAgent/HeatEmissionAgent/src/main/java/uk/ac/cam/cares/jps/agent/heat/EndpointConfig.java (hunk 1 above)
Comment (nagarajankarthik) [reply Agents/HeatEmissionAgent/HeatEmissionAgent/src/main/java/uk/ac/cam/cares/jps/agent/heat/EndpointConfig.java:18]: Done. Thanks for the recommendation.

File: Agents/HeatEmissionAgent/HeatEmissionAgent/src/main/java/uk/ac/cam/cares/jps/agent/heat/HeatEmissionAgent.java (hunk 2 above)
Comment (nagarajankarthik) [reply Agents/HeatEmissionAgent/HeatEmissionAgent/src/main/java/uk/ac/cam/cares/jps/agent/heat/HeatEmissionAgent.java:60]: The pharmaceutical, food, semiconductors and chemicals plants data are stored in the factories table. The data for the precision engineering, printing and data centres are stored in separate tables. The data for the various industrial facilities is grouped into PostgreSQL tables based on the parameters used to estimate heat emissions. 

The factories table contains data for facilities in multiple industries because the parameters used to estimate heat emissions are the specific energy conusmption and production volume in all cases.

File: Agents/HeatEmissionAgent/README.md (hunk 3 above)
Comment (nagarajankarthik) [reply Agents/HeatEmissionAgent/README.md:3]: The classes 'Factory' and 'DataCentre' are subclasses of 'IndustrialFacility'.

File: Agents/HeatEmissionAgent/README.md (hunk 4 above)
Comment (nagarajankarthik) [reply Agents/HeatEmissionAgent/README.md:55]: It should be industrial facilities. I have made this correction.

File: Agents/HeatEmissionAgent/README.md (hunk 5 above)
Comment (nagarajankarthik) [reply Agents/HeatEmissionAgent/README.md]: Done.

**Review by sganguly-cares — APPROVED** [review]
